api/capi/src/nntrainer.cpp

```diff
@@ -28,10 +28,42 @@
 #include <stdarg.h>
 #include <string.h>
 
+/**
+ * @brief     function to wrap an exception to predefined error value
+ * @param[in] func must be wrapped inside lambda []() -> int otherwise compile
+ * error will be raised
+ * @retval    errorno
+ */
+template <typename F> static int nntrainer_exception_boundary(F &&func) {
+  try {
+    return func();
+  } catch (std::invalid_argument &e) {
+    ml_loge("%s %s", typeid(e).name(), e.what());
+    return ML_ERROR_INVALID_PARAMETER;
+  } catch (std::range_error &e) {
+    ml_loge("%s %s", typeid(e).name(), e.what());
+    return ML_ERROR_INVALID_PARAMETER;
```

Comment:
ML_ERROR_RESULT_OUT_OF_RANGE??

Reply:
https://github.com/nnstreamer/nntrainer/blob/bb2dd7e383eda96d9502683b76c55ce72f9659f2/api/capi/include/nntrainer.h#L62-L72

I actually reverted that change because it was not defined and I was not sure if it is okay to change `ml_error_e`.
@jijoongmoon  could we add another ML_ERROR to `capi`?

Reply:
uhh, it seems like nntrainer_error.h and nntrainer.h are not in sync.

Reply:
This is for the tizen api so have to sync accrodingly.

Reply:
You are breaking the compatibility with Tizen ML API sets here.

It's time to consider using nnstreamer-capi-devel.rpm to get error-code header files (ml-api-common.h)

Reply:
Added #243 to resolve this issue.